### PR TITLE
update feature/gdas-validation to work with more recent GDASApp and g-w

### LINKED ISSUE #107: Update feature/gdas-validation to be consistent with g-w and GDASApp development
JEDI-T2O `feature/gdas-validation` has fallen behind g-w `develop` and GDASApp `feature/gdas-validation`.  This issue is opened to bring `feature/gdas-validation` up to date with the current  (as of 1/18/2024) state of g-w and GDASApp.

### PR DESCRIPTION
**Pull request by RussTreadon-NOAA**
Changes in GDASApp and global-workflow require updates to `feature/gdas-validation`.  This PR includes the changes needed for developers to install, build, and configure an an environment for GDAS validation tests with more recent snapshots of GDASApp and global-workflow.

Resolves #107

### CODE REVIEW

**Review by ADCollard — APPROVED** [review]
Tested through the build step on Orion with no issues.  I will test full end-to-end, but approving.

**Comment by RussTreadon-NOAA** [comment]
Thank you, @ADCollard , for the review and approval.  A [comment](https://github.com/NOAA-EMC/JEDI-T2O/issues/107#issuecomment-1900771671) describing results from an Orion GSI and GDASApp test are include in issue #107.

**Comment by RussTreadon-NOAA** [comment]
**Special Note**  

This PR changes the `-c` option of `setup_workspace.sh`.  

The hash of GDASApp which g-w `develop` points at is cloned into `sorc/gdas.cd`.  As of this PR, g-w `develop` clones GDASApp @ `b79c0a2`.  

Previously the head of GDASApp branch `feature/gdas-validation` was checked out.  This is _**no longer true**_ with the proposed changes to `setup_workspace.sh`.  

This PR does not alter the g-w or GSI clones.  g-w `develop` is still cloned.  GSI branch `feature/gdas-validation` is still checked out.

**Comment by ADCollard** [comment]
> One added note: This PR changes the `-c` option of `setup_workspace.sh`. The head of GDASApp `develop` is used. GDASApp branch `feature/gdas-validation` is _**not**_ checked out.
> 
> This PR does not alter the g-w or GSI clones. g-w `develop` is still cloned. GSI branch `feature/gdas-validation` is still checked out.

@RussTreadon-NOAA Is that correct?  When I ran `setup_workspace.sh -c` I got a specific tag (not develop) for GDASApp.  I believe that is the expected behaviour?

```
Orion-login-2[19] acollard$ git branch
* (HEAD detached at b79c0a2)
  develop
```

**Comment by RussTreadon-NOAA** [comment]
@ADCollard , you are correct.  With g-w's move to submodules, we get specific hashes of various components.  `b79c0a2` is the hash for a specific commit to GDASApp `develop`.  [`gdas.cd`](https://github.com/NOAA-EMC/global-workflow/tree/develop/sorc) in g-w `develop` currently points at `b79c0a2`.  

If a developer wants a different GDASApp branch or hash for GDAS validation, they must `cd sorc/gdas.cd` and execute `git checkout` for the hash or branch they want.

I'll update my comment to clarify this important detail.  Thank you catching my mistake.

File: GDAS-validation/gdas_config/config.atmanl [hunk 1]
```diff
@@ -6,10 +6,9 @@
 echo "BEGIN: config.atmanl"
 
 export CASE_ANL=${CASE}
-#export CASE_ANL=C384
```
Comment (emilyhcliu) [inline GDAS-validation/gdas_config/config.atmanl:9]: @RussTreadon-NOAA Can we run C768 for atmanl now?

Comment (RussTreadon-NOAA) [reply GDAS-validation/gdas_config/config.atmanl:9]: Yes, we can run the minimizer on C768 for certain combinations of `[layout_x, layout_y]`.  `config.resources` in this PR sets `[layout_x, layout_y]` = `[8, 12]`.  I've run `fv3jedi_var.x` with `CASE_ANL=C768` using the `[8, 12]` layout.

Comment (emilyhcliu) [reply GDAS-validation/gdas_config/config.atmanl:9]: @RussTreadon-NOAA Thanks for your explanation.

**Review by emilyhcliu — APPROVED** [review]

**Comment by RussTreadon-NOAA** [comment]
**Hercules Test**
Exercise `setup_workspace.sh` on Hercules.  The `-c` open worked without error.  The GSI build under the `-b` option fails.  The GDASApp build runs to completion.

GDAS-validation uses GSI branch `feature/gdas-validation`.  The Hera and Orion modulefiles are modified to use Emily's CRTM build.  This modification was not present in the `feature/gdas-validation` snapshot of `modulefiles/gsi_hercules.lua`.   Adding the following to `modulefiles/gsi_hercules.lua`
```
unload("crtm/2.4.0")
setenv("crtm_ROOT","/work/noaa/da/eliu/JEDI-GDAS/crtm_v2.4.1-jedi.1-intel2022/build")
setenv("crtm_VERSION","2.4.1-jedi.1")
setenv("CRTM_INC","/work/noaa/da/eliu/JEDI-GDAS/crtm_v2.4.1-jedi.1-intel2022/build/module")
setenv("CRTM_LIB","/work/noaa/da/eliu/JEDI-GDAS/crtm_v2.4.1-jedi.1-intel2022/build/lib/libcrtm_static.a")
setenv("CRTM_FIX","/work2/noaa/da/cmartin/GDASApp/fix/crtm/2.4.0")
```
is not sufficient.  The build aborts with the following message
```
CMake Error at /work/noaa/da/eliu/JEDI-GDAS/crtm_v2.4.1-jedi.1-intel2022/build/crtm-import.cmake:11 (message):
  Package crtm provides Fortran modules built with Intel-2021.5.0.20211109
  but this build for gsi uses incompatible compiler Intel-2021.9.0.20230302
Call Stack (most recent call first):
  /work/noaa/da/eliu/JEDI-GDAS/crtm_v2.4.1-jedi.1-intel2022/build/crtm-config.cmake:69 (include)
  src/gsi/CMakeLists.txt:75 (find_package)
```

If we want to build GSI branch `feature/gdas-validation` on Hercules using the same CRTM as the GDASApp build, we need a Hercules build of `eliu/JEDI-GDAS/crtm_v2.4.1-jedi.1`.

Alternatively, we can modify GSI `modulefiles/gsi_hercules.lua` to use the CRTM built as part of the GDASApp.  This gets complicated.  We need to build GDASApp first and then edit `modulefiles/gsi_hercules.lua` to point at the GDASApp crtm build.  Perhaps we can simplify this by having `modulefiles/gsi_hercules.lua` use environment variables to point at the correct CRTM pieces.  

I was able to build `gsi.x` and `enkf.x` on Hercules using my GDASApp crtm build.  Additional changes are needed to run `gsi.x` within the GDAS validation environment.  

A modified `config.anal` from JEDI-T2O is copied into the GDAS validation config directory as part of `setup_workspace.sh -b`.  This `config.anal` has Orion and Hera blocks to pick up the correct `CRTM_FIX` and CRTM library.  Logic needs to be added to `config.anal` to set these variables for Hercules.  I made these changes in a local copy of `config.anal`.   After this, gdasanal ran to completion on Hercules from within the GDAS validation environment.

How do we want to handle this long term?  

Using the GDASApp build of CRTM has the benefit of ensuring CRTM consistency between `gsi.x` and `fv3jedi_var.x`.  This, however, entails on the fly modification of `gsi_hercules.lua` and `config.anal` - not hard but we need to make sure we get it right.   We can side step these on the fly modifications by installing the desired CRTM in a central location on Hercules and pointing `gsi_hercules.lua` and `config.anal` at the centrally intsalled snapshot.  We use the latter approach on Hera and Orion.  

Either way, I don't think Hercules modifications need to be in this PR.  Extension to Hercules could be the subject of a new issue and PR.  Shall we merge this PR into JEDI-T2O `feature/gdas-validation` in it's current state?

What do you think @CoryMartin-NOAA , @ADCollard , and @emilyhcliu ?

**Comment by CoryMartin-NOAA** [comment]
I would vote for merging in as is and then deciding what to do with Hercules later. My $0.02 is that if we are doing 1:1 validation for UFO, then the exact CRTM is important, but if we are using Hercules for cycling or other testing/validation, then CRTM can be 'close enough'.

**Comment by RussTreadon-NOAA** [comment]
Thank you @CoryMartin-NOAA , for your reply.

**Comment by ADCollard** [comment]
@ADCollard Another vote for merging and addressing Hercules later.   I would suggest ensuring that all builds are consistent, so we should have a Hercules build of `eliu/JEDI-GDAS/crtm_v2.4.1-jedi.1`.  I defer to @emilyhcliu on how best to do this.

**Comment by emilyhcliu** [comment]
@RussTreadon-NOAA @CoryMartin-NOAA @ADCollard 
I also vote for merging it now and dealing with the CRTM issue for Hercules later. 

It looks like we have plan to cycle on Hercules.  I just tried Hercules and I can log in without problem.
I will go ahead and build CRTM_2.4.1-jedi.1 on Hercules

**Comment by emilyhcliu** [comment]
> @RussTreadon-NOAA @CoryMartin-NOAA @ADCollard I also vote for merging it now and dealing with the CRTM issue for Hercules later.
> 
> It looks like we have plan to cycle on Hercules. I just tried Hercules and I can log in without problem. I will go ahead and build CRTM_2.4.1-jedi.1 on Hercules

First time on Hercules.    I just found that the disk spaces (work, work2, ..etc) are shared between Hercules and Orion.

**Comment by RussTreadon-NOAA** [comment]
Thank you @emilyhcliu , @ADCollard , and @CoryMartin-NOAA for your replies.  I'll merge this PR into `feature/gdas-validation`.  

I found that g-w `develop` is not fully ready to run GDASApp on Hercules.  I'll open a g-w issue and PR to get a fix into g-w `develop`.   Once the hercules crtm is ready, we can update `gsi_hercules.lua` in GSI branch `feature/gdas-validation`.

**Pull request merged**